store/models/run_test.go

```diff
@@ -117,35 +119,87 @@ func TestRunResult_Value(t *testing.T) {
 	}
 }
 
-func TestRunResult_MergeData(t *testing.T) {
+func TestRunResult_Merge(t *testing.T) {
 	t.Parallel()
 
+	nullString := cltest.NullString(nil)
+	jrID := utils.NewBytes32ID()
 	tests := []struct {
-		name        string
-		input       string
-		want        string
-		wantErrored bool
+		name            string
+		originalData    string
+		originalError   null.String
+		originalPending bool
+		originalJRID    string
+		inData          string
+		inError         null.String
+		inPending       bool
+		inJRID          string
+		wantData        string
+		wantError       null.String
+		wantPending     bool
+		wantJRID        string
 	}{
-		{"replace field", `{"value":"new hotness"}`,
-			`{"value":"new hotness"}`, false},
-		{"add field", `{"extra":1}`,
-			`{"value":"old and busted","extra":1}`, false},
-		{"replace and add field", `{"value":"new hotness","extra":1}`,
-			`{"value":"new hotness","extra":1}`, false},
+		{"merging data",
+			`{"value":"old&busted","unique":"1"}`, nullString, false, jrID,
+			`{"value":"newHotness","and":"!"}`, nullString, false, jrID,
+			`{"value":"newHotness","unique":"1","and":"!"}`, nullString, false, jrID},
+		{"original error",
+			`{"value":"old"}`, cltest.NullString("new problem"), false, jrID,
+			`{}`, nullString, false, jrID,
+			`{"value":"old"}`, cltest.NullString("new problem"), false, jrID},
```

Comment:
When would we ever merge a `RunResult` and want to keep the old error? When would we ever merge a `RunResult` after it has errored?